### ISSUE TITLE
WampSharp appears to return wrong error information on failed deserialisation

### ISSUE THREAD

**Issue by Capsup** [Code-Sharp/WampSharp#169]
Hi,

I've been working on a backend system that uses WampSharp as the client to register RPCs and publish events. I've run into a problem where if a request is made to the client through the crossbar.io router with a correct procedure name but with an incorrect payload (I.E. the function expects a dictionary but is given something that can't be deserialised to one), WampSharp will return an WampException. However, crossbar.io appears to expect a Dictionary<string,string> Details for a WampException.

So whenever WampSharp returns an exception without that, the following error happens in crossbar:
```2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.router.Router] Validate 'call' for 'erp.report.get'

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.role.RouterRole] CrossbarRouterRoleStaticAuth.authorize backend erp.report.get call

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.router.Router] Authorized action 'call' for URI 'erp.report.get' by session 3984500893468939 with authid 'AJTV-7LJG-AHET-ESF7-H7RR-Q6EA' and authrole 'backend' -> authorization: {'cache': True, 'disclose': False, 'allow': True}

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.protocol.WampWebSocketServerProtocol] Traceback (most recent call last):
  File "\python\python35-32\lib\site-packages\autobahn\wamp\websocket.py", line 88, in onMessage
    for msg in self._serializer.unserialize(payload, isBinary):
  File "\python\python35-32\lib\site-packages\autobahn\wamp\serializer.py", line 131, in unserialize
    msg = Klass.parse(raw_msg)
  File "\python\python35-32\lib\site-packages\autobahn\wamp\message.py", line 1006, in parse
    details = check_or_raise_extra(wmsg[3], u"'details' in ERROR")
  File "\python\python35-32\lib\site-packages\autobahn\wamp\message.py", line 207, in check_or_raise_extra
    raise ProtocolError(u"{0}: invalid type {1}".format(message, type(value)))
autobahn.wamp.exception.ProtocolError: 'details' in ERROR: invalid type <class 'NoneType'>


2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.protocol.WampWebSocketServerProtocol] Failing WAMP-over-WebSocket transport: code=1002, reason="WAMP Protocol Error ('details' in ERROR: invalid type <class 'NoneType'>)"

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.protocol.WampWebSocketServerProtocol] failing connection: 1002: WAMP Protocol Error ('details' in ERROR: invalid type <class 'NoneType'>)

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.protocol.WampWebSocketServerProtocol] Connection to/from tcp4:81.7.186.84:61192 was closed cleanly

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.protocol.WampWebSocketServerProtocol] _connectionLost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly: Connection done.
]

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.protocol.WampWebSocketServerProtocol] WAMP-over-WebSocket transport lost: wasClean=False, code=1006, reason="connection was closed uncleanly (peer dropped the TCP connection without previous WebSocket closing handshake)"

2017-01-06T15:18:56+0100 [Router       4512 crossbar.router.dealer.Dealer] Cancelling in-flight INVOKE with id=2607698564157988 on session 1898585967313183
```

The same happens if I attempt to throw an exception like so:
```csharp
throw new WampException( "erp.scandoc.todo.error.invalidid", $"ERROR! No todo with the id: '{ id }'" );
```
instead, I have to throw exceptions like so:
```csharp
throw new WampException( new Dictionary<string, object>(), "erp.scandic.todo.error.invalidid", $"ERROR! No todo with the id: '{ id }'", null );
```

Is this an issue that has cropped up with crossbar.io, or has the specification for the ERROR message changed? As far as I can see [here](https://github.com/wamp-proto/wamp-proto/blob/master/rfc/text/basic/bp_messages.md), the ERROR message MUST return a dictionary for details?

**Comment by darkl** [comment]
Didn't I fix this already? See issue #162.
What version are you using? Did you try latest version.

**Comment by Capsup** [comment]
Yes, you're right, my bad for not looking at new versions. It definitely works as expected now. Thanks alot.
